examples/configurationdemo/source/c/src/kaa_demo.c

```diff
@@ -37,112 +36,37 @@
-                                             , KAA_SERVICE_USER
-                                             , KAA_SERVICE_EVENT
-                                             , KAA_SERVICE_LOGGING
                                              , KAA_SERVICE_CONFIGURATION };
 static const int OPERATIONS_SERVICES_COUNT = sizeof(OPERATIONS_SERVICES) / sizeof(kaa_service_t);
 
+
 static kaa_transport_channel_interface_t bootstrap_channel;
 static kaa_transport_channel_interface_t operations_channel;
 
-static void *log_storage_context         = NULL;
-static void *log_upload_strategy_context = NULL;
-
-static bool is_shutdown = false;
-
-
-
-/* forward declarations */
-
-extern kaa_error_t ext_unlimited_log_storage_create(void **log_storage_context_p
-                                                    , kaa_logger_t *logger);
-
-extern kaa_error_t ext_log_upload_strategy_by_volume_create(void **strategy_p
-                                                          , kaa_channel_manager_t   *channel_manager
-                                                          , kaa_bootstrap_manager_t *bootstrap_manager);
-extern kaa_error_t ext_log_upload_strategy_by_volume_set_threshold_count(void *strategy, size_t threshold_count);
-
-
-
-/*
- * Initializes the Kaa log collector.
- */
-kaa_error_t kaa_log_collector_init()
-{
-    kaa_error_t error_code = ext_unlimited_log_storage_create(&log_storage_context, kaa_context_->logger);
-    if (error_code) {
-        KAA_LOG_ERROR(kaa_context_->logger, error_code, "Failed to create log storage");
-        return error_code;
-    }
-
-    error_code = ext_log_upload_strategy_by_volume_create(&log_upload_strategy_context
-                                                        , kaa_context_->channel_manager
-                                                        , kaa_context_->bootstrap_manager);
-
-    if (error_code) {
-        KAA_LOG_ERROR(kaa_context_->logger, error_code, "Failed to create log upload strategy");
-        return error_code;
-    }
-
-    error_code = ext_log_upload_strategy_by_volume_set_threshold_count(log_upload_strategy_context
-                                                                     , KAA_DEMO_UPLOAD_COUNT_THRESHOLD);
-
-    error_code = kaa_logging_init(kaa_context_->log_collector
-                                , log_storage_context
-                                , log_upload_strategy_context);
-
-    return error_code;
-}
-
-
 void kaa_demo_print_configuration_message(const kaa_root_configuration_t *configuration)
 {
-    if (configuration->message->type == KAA_CONFIGURATION_UNION_STRING_OR_NULL_BRANCH_1) {
-        KAA_LOG_TRACE(kaa_context_->logger, KAA_ERR_NONE, "Current configuration: null");
+    if (configuration->address_list->type == KAA_CONFIGURATION_UNION_ARRAY_LINK_OR_NULL_BRANCH_0) {
+        printf("Configuration body:\n");
+        kaa_list_t *list_of_links = (kaa_list_t*) configuration->address_list->data;
```

Comment:
I suggest more convenient way to deal with the kaa_list_t type.

kaa_configuration_link_t* current_link = NULL;
kaa_list_t *list_of_links = (kaa_list_t*) configuration->address_list->data;
while (list_of_links) {
    current_link = (kaa_configuration_link_t*) kaa_list_get_data(list_of_links);
    printf("%s - %s\n", current_link->label->data,current_link->url->data);
    list_of_links = kaa_list_next(list_of_links);
}